packages/xstate-store/src/react.ts

```diff
@@ -94,3 +100,48 @@ export const useStore: {
 
   return storeRef.current;
 };
+
+/**
+ * A React hook that subscribes to the `atom` and returns the current value of
+ * the atom.
+ *
+ * @example
+ *
+ * ```ts
+ * const atom = createAtom(0);
+ *
+ * const Component = () => {
+ *   const count = useAtom(atom);
+ *
+ *   return (
+ *     <div>
+ *       <div>{count}</div>
+ *       <button onClick={() => atom.set((c) => c + 1)}>Increment</button>
+ *     </div>
+ *   );
+ * };
+ * ```
+ *
+ * @param atom The atom, created from `createAtom(…)`
+ * @param selector An optional function which takes in the `snapshot` and
+ *   returns a selected value
+ * @param compare An optional function which compares the selected value to the
+ *   previous value
+ */
+export function useAtom<TAtom extends AnyAtom>(
+  atom: TAtom
+): SnapshotFromAtom<TAtom>;
+export function useAtom<TAtom extends AnyAtom, T>(
+  atom: TAtom,
+  selector: (snapshot: SnapshotFromAtom<TAtom>) => T,
+  compare?: (a: T | undefined, b: T) => boolean
```

Comment:
To make it easier to work with this we should aim for this signature
```suggestion
  compare?: (a: T, b: T) => boolean
```

The first selected item doesn't ever have to be compared against the previous value - we know the previous doesn't exist